### ISSUE TITLE
Add theme / color changing support in running instance

### ISSUE THREAD

**Issue by gustafj** [kovidgoyal/kitty#431]
When presenting on an external monitor, or working in highly lit areas (e.g. outdoors) I often find myself using a bright color palett, as opposed to the dark I use in my day-to-day work.
Currently this requires manual modification of the `foreground` and `background` config setting along with possible changes of the `colorX` setting followed by starting a fresh instance of kitty and all applications that were running in the old instance.

To mitigate this manual work and to be able to preserv the currently active applications and their history, I would like to be able to create a few (2+) pre defined "themes" in the kitty.conf and then be able to switch between them using either the kitty API or key-combination.

E.g:
```
# Active theme
theme Solarized-Dark 

# Theme management
map ctrl+shift+f7 set_theme Solarized-Dark
map ctrl+shift+f8 set_theme Solarized-Light
map ctrl+shift+f9 previous_theme
map ctrl+shift+f10 next_theme

# Theme definition blocks
theme Solarized-Dark {
  foreground  #839496
  background #002b36

  color0   #073642 # black
  color8   #002b36
  color1   #dc322f # red
  color9   #cb4b16
  color2   #859900 # green
  color10  #586e75
  color3   #b58900 # yellow
  color11  #657b83
  color4   #268bd2 # blue
  color12  #839496
  color5   #d33682 # magenta
  color13  #6c71c4
  color6   #2aa198 # cyan
  color14  #93a1a1
  color7   #eee8d5 # white
  color15  #fdf6e3
}
 
theme Solarized-Light {
  foreground  #657b83
  background #fdf6e3

  color0   #073642 # black
  color8   #002b36
  color1   #dc322f # red
  color9   #cb4b16
  color2   #859900 # green
  color10  #586e75
  color3   #b58900 # yellow
  color11  #657b83
  color4   #268bd2 # blue
  color12  #839496
  color5   #d33682 # magenta
  color13  #6c71c4
  color6   #2aa198 # cyan
  color14  #93a1a1
  color7   #eee8d5 # white
  color15  #fdf6e3
}
 
theme Original {
  foreground  #dddddd
  background #000000

  color0   #000000 # black
  color8   #4d4d4d
  color1   #cc0403 # red
  color9   #f2201f
  color2   #19cb00 # green
  color10  #23fd00
  color3   #cecb00 # yellow
  color11  #fffd00
  color4  #0d73cc # blue
  color12 #1a8fff
  color5   #cb1ed1 # magenta
  color13  #fd28ff
  color6   #0dcdcd # cyan
  color14  #14ffff
  color7   #dddddd # white
  color15  #ffffff
}

theme Xterm {
  foreground  #e4e4e4
  background #262626

  color0   #262626 # black
  color8   #1c1c1c
  color1   #d70000 # red
  color9   #d75f00
  color2   #5f8700 # green
  color10  #585858
  color3   #af8700 # yellow
  color11  #626262
  color4   #0087ff # blue
  color12  #808080
  color5   #af005f # magenta
  color13  #5f5faf
  color6   #00afaf # cyan
  color14  #8a8a8a
  color7   #e4e4e4 # white
  color15  #ffffd7
}
```

**Comment by kovidgoyal** [comment]
You can switch all colors using standard terminal escape codes, that work in most terminal emulators. A bit of googling will find you information about them.

**Comment by gustafj** [comment]
I was not aware that it was possible to change the **default** colors (as in including **all** history that does not define a specific color) using escape codes, *and the documentation for it was not that obvious*.
After a bit of digging I found it in the *"Operating System Controls"* part of the `xterm` docs.

For reference if anyone else has the same needs as I:
*Note: I only change the foreground and background colors of default text & selection since solarized uses the same template regardless if its dark or light, if a specific color needs to be changed as well that should be possible using `printf '\e]4;<color>;<value>\e\\'` but I have not tested that.*
```
# For solarized light
# Foreground
printf '\e]10;#657b83\e\\'
# Background
printf '\e]11;#fdf6e3\e\\'
# Highlight (selection) background
printf '\e]17;#657b83\e\\'
# Highlight (Selection) foreground
printf '\e]19;#fdf6e3\e\\'

# For solarized dark
# Foreground
printf '\e]10;#839496\e\\'
# Background
printf '\e]11;#002b36\e\\'
# Highlight (selection) background
printf '\e]17;#839496\e\\'
# Highlight (Selection) foreground
printf '\e]19;#002b36\e\\'
```

To summarize its not quite what I wanted as it only changes the current tab and it does not stay after a `reset(1)`, but apart from that it does indeed change the colors in the way I want.
So *theme* support added to `kitty` that changes the colors of all tabs and stays after a `reset` would be preferrable IMHO.

**Comment by SolitudeSF** [comment]
to apply it to every terminal write this sequence to every file in /dev/pts/[0-9]*

**Comment by kovidgoyal** [comment]
You'd have to do that anyway, since the final color is always overridden by the color set via escape code. Changing the configured color in kitty wont actually change colors if they have been overridden by escape code colors. As for reset, simply alias reset to do a reset and then run your theme setting script. Similarly in your shell's startup file have the theme setting script run at startup. So all new terminals get their colors set according to your theme script. 

Although, I have to say I dont know how you'd make themeing work with full screen terminal apps that often have their own color schemes. You'd need to configure each one individually. 

Oh and there is also an escape code to set the cursor color, which you missed, IIRC it is 12

**Comment by gustafj** [comment]
@SolitudeSF Thanks, works great.
@kovidgoyal Thanks for the support, got a setup now that works mostly as I want.

Of course the full screen apps that define their own color schemes would need personalized settings, and perhaps a custom script is better in that sense (one place for all settings, terminal, apps, etc). Just that I was used to changing the color palette (including background & foreground) for the terminal editor itself. E.g using `Edit -> Profile preferences -> Colors -> Built-in schemes` in Gnome Terminal.
I mostly use the terminal editor without full screen apps, so that is not much of an issue to me.

These escape codes works in Gnome Terminal as well, so now I have a seamless setup with identical colors in both kitty and Gnome Terminal (though I'm only ever using Gnome Terminal for reference checks...).

*PS. I did not really miss the cursor escape, just that I had no need to change it ;) But others might indeed like to know its possible.*

**Comment by kovidgoyal** [comment]
Yeah, that's why I usually recommend people use the escape codes, it works across most terminals and is a single place to define things.
